internal/inspect/inspect.go

```diff
@@ -1,104 +1,123 @@
 package inspect
 
 import (
+	"encoding/json"
 	"fmt"
 	"io"
+	"os"
 	"sort"
 	"strings"
 	"text/tabwriter"
 
 	"github.com/deislabs/cnab-go/bundle"
+	"github.com/docker/app/internal"
 	"github.com/docker/app/render"
 	"github.com/docker/app/types"
+	"github.com/docker/app/types/metadata"
 	"github.com/docker/app/types/parameters"
 	composetypes "github.com/docker/cli/cli/compose/types"
 	units "github.com/docker/go-units"
 )
 
+type service struct {
+	Name     string
+	Image    string
+	Replicas int
+	Mode     string
+	Ports    string
+}
+
+type attachment struct {
+	Path string
+	Size int64
+}
+
+type appInfo struct {
+	Metadata       metadata.AppMetadata
+	Services       []service
+	Networks       []string
+	Volumes        []string
+	Secrets        []string
+	parametersKeys []string
+	Parameters     map[string]string
+	Attachments    []attachment
+}
+
 // Inspect dumps the metadata of an app
 func Inspect(out io.Writer, app *types.App, argParameters map[string]string, imageMap map[string]bundle.Image) error {
+	outputFormat := os.Getenv(internal.DockerInspectFormatEnvVar)
```

Comment:
nit: Move this closer to where you are actually using it?

Reply:
Done